dashboard/test/models/reference_guide_test.rb

```diff
@@ -0,0 +1,51 @@
+require 'test_helper'
+
+class ReferenceGuideTest < ActiveSupport::TestCase
+  test "can create reference guide" do
+    reference_guide = create :reference_guide
+    assert reference_guide.id
+  end
+
+  test "do not allow invalid keys" do
+    create :reference_guide, key: 'valid.KEY_'
+    assert_raises ActiveRecord::RecordInvalid do
+      create :reference_guide, key: 'fgsfds.'
+    end
+    assert_raises ActiveRecord::RecordInvalid do
+      create :reference_guide, key: '.fgsfds'
+    end
+    assert_raises ActiveRecord::RecordInvalid do
+      create :reference_guide, key: '\\ $ % * & @'
+    end
+  end
+
+  test "reference guides are unique by key in course version" do
+    create :reference_guide, key: 'page', course_version_id: 1
+    create :reference_guide, key: 'page', course_version_id: 2
+    assert_raises ActiveRecord::RecordInvalid do
+      create :reference_guide, key: 'page', course_version_id: 1
```

Comment:
it is not great to use ids which might not point to valid course version objects, for example this could break later if we decided to add a validation for ref guide which tested some property on the course version. a better practice would be to have the course_version_id point to actual course versions, allowing the factories to create said course version when possible. here, that could translate to:
```suggestion
    ref1 = create :reference_guide, key: 'page'
    create :reference_guide, key: 'page'
    assert_raises ActiveRecord::RecordInvalid do
      create :reference_guide, key: 'page', course_version_id: ref1.course_version
```